File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
//hello
        Map<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];

            if (map.containsKey(complement)) {
                return new int[]{ map.get(complement), i };
            }

            map.put(nums[i], i); // put AFTER checking
        }

        return new int[]{}; // only if no solution found
    }
}

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        paired = []
        for idx in range(len(nums)):
            value = nums[idx]
            paired.append([value,idx])
        paired.sort()

        left = 0
        right = len(paired) -1 

        while left < right :
            left_value = paired[left][0]
            right_value = paired[right][0]
            curr_sum = left_value + right_value

            if curr_sum == target:
                left_index = paired[left][1]
                right_index = paired[right][1]
                result = [left_index , right_index ]
                result.sort()
                return result
            elif curr_sum < target:
                left += 1
            else:
                right -= 1
        return []
